.gitignore

```diff
@@ -1,3 +1,9 @@
 /vendor
 /composer.lock
 /.phpunit.result.cache
+
+# https://stackoverflow.com/a/16318111
+/tools/*
+!/tools/ecs
+/tools/ecs/*
+!/tools/ecs/composer.json
```

Comment:
Unfortunately not, I've put a stackoverflow link above to "document" why it's required: https://stackoverflow.com/a/16318111

Reply:
I've tested both version too on my hand and the result is quite different:

![image](https://user-images.githubusercontent.com/1524501/174002651-8ac2ed42-65ed-4c50-a656-90a64ac38e39.png)

![image](https://user-images.githubusercontent.com/1524501/174002902-15df3e3e-c281-403d-87a6-b661d47cee5a.png)

I thought it could be a git version problem but upgrading from `2.25` to `2.36` didn't change the behavior.

What makes you think they work equally fine ?

Reply:
Weird, I can't reproduce your issue here. When I remove these 2 lines, git still tracks `composer.json` and `ecs.php`.
I guess it doesn't hurt much to leave them, though.